src/export/bops/index.ts

```diff
@@ -0,0 +1,536 @@
+            try {
+              // skip returning internal _contact data object, just return main key values
+              const contactObject = Object.values(bc.data!).filter(
+                (x) => typeof x === "string"
+              );
+              return [Object.values(contactObject).join(" ")];
+            } catch (err) {
+              return [JSON.stringify(bc.data)];
+            }
+          case ComponentType.DateInput:
+          case ComponentType.NumberInput:
+          case ComponentType.TextInput:
+            return Object.values(bc.data ?? {}).map((x) => String(x));
+          case ComponentType.Checklist:
+          case ComponentType.Question:
+          default:
+            return bc.answers ?? [];
+        }
+      })();
+
+      const responses = answers.map((id) => {
+        let value = id;
+        const metadata: ResponseMetaData = {};
+
+        if (flow[id]) {
+          // XXX: this is how we get the text representation of a node until
+          //      we have a more standardised way of retrieving it. More info
+          //      https://github.com/theopensystemslab/planx-new/discussions/386
+          value = flow[id].data?.text ?? flow[id].data?.title ?? "";
+
+          if (flow[id].data?.flag) {
+            const flag = flatFlags.find((f) => f.value === flow[id].data?.flag);
+            if (flag) {
+              metadata.flags = [`${flag.category} / ${flag.text}`];
+            }
+          }
+        }
+
+        const ob: Response = { value };
+        if (Object.keys(metadata).length > 0) ob.metadata = metadata;
+        return ob;
+      });
+
+      const ob: QuestionAndResponses = {
+        question: question?.data?.text ?? question?.data?.title ?? "",
+        responses,
+      };
+
+      let metadata: QuestionMetaData = {};
+      if (bc.auto) metadata.auto_answered = true;
+      if (flow[id]?.data?.policyRef) {
+        metadata.policy_refs = [
+          // remove html tags
+          { text: flow[id].data?.policyRef?.replace(/<[^>]*>/g, "").trim() },
+        ];
+      }
+      metadata = addPortalName(id, flow, metadata);
+      // metadata = addSectionName(id, metadata);
+
+      if (Object.keys(metadata).length > 0) ob.metadata = metadata;
+
+      return ob;
+    })
+    .filter(Boolean) as Array<QuestionAndResponses>;
+
+  return { proposal_details, feedback };
+};
+
+export function getBOPSParams({
+  breadcrumbs,
+  flow,
+  passport,
+  sessionId,
+  flowName,
+}: {
+  breadcrumbs: Breadcrumbs;
+  flow: FlowGraph;
+  passport: Passport;
+  sessionId: string;
+  flowName: string;
+}) {
+  const data = {} as BOPSFullPayload;
+  data.application_type = DEFAULT_APPLICATION_TYPE;
+
+  // Overwrite default application type if this isn't an LDC (relies on LDC flows having consistent slug)
+  //   eg because makeCsvData which is used across services calls this method
+  if (flowName && flowName !== "Apply for a lawful development certificate") {
```

Comment:
your right that `flowName` is required so this line was a bit redundant - all we really need to check here is what the flow name _is_: if it's LDC then use DEFAULT_APPLICATION_TYPE, otherwise set as flowName. 

supporting a value other than DEFAULT_APPLICATION_TYPE makes this function callable to generate CSV data that sharse the same data structure as BOPS but are used in non-LDC contexts.